pkg/operator/starter.go

```diff
@@ -165,6 +165,22 @@ func RunOperator(ctx context.Context, cc *controllercmd.ControllerContext) error
 			return isVSphere
 		},
 		nil,
+	).WithConditionalResources(
+		bindata.Asset,
+		[]string{
+			"assets/kube-controller-manager/gce/cloud-provider-role.yaml",
+			"assets/kube-controller-manager/gce/cloud-provider-binding.yaml",
+			"assets/kube-controller-manager/sa.yaml",
+		},
+		func() bool {
+			// We do not want to apply these resources, so must return false here
+			return false
+		},
+		func() bool {
+			// The resources above are required for the 4.14 -> 4.15 upgrade path.
```

Comment:
are you planning to backport this to 4.15 branch?

Reply:
No, this would break 4.14 to 4.15 upgrades, that's why we are doing this now. The file needs to be removed only once the upgrade to 4.15 is complete, it cannot be removed mid update as that could cause KCM to fall over and halt the upgrade.

Reply:
I mean, 4.15 is not released yet, so I think it could still work if we do it before the release.

Reply:
^ if the RBAC is not used in the 4.14

Reply:
I've been working on the basis that RBAC is used in 4.14, that's why we are removing now, but I will double check when we stopped using it.

Reply:
Yep, confirmed, RBAC is used in 4.14, not used in 4.15, so I want to remove only from 4.16 to avoid breaking upgrades

Reply:
Thanks for looking into this. Ok let's remove this in 4.16 then.